support-workers/src/typescript/lambdas/createPaymentMethodLambda.ts

```diff
@@ -0,0 +1,242 @@
+import { combinedAddressLine } from '../model/address';
+import type {
+	DirectDebitPaymentFields,
+	PaymentFields,
+	PayPalPaymentFields,
+	StripeHostedPaymentFields,
+	StripePaymentFields,
+} from '../model/paymentFields';
+import type {
+	DirectDebitPaymentMethod,
+	PaymentMethod,
+	PayPalPaymentMethod,
+	StripePaymentMethod,
+} from '../model/paymentMethod';
+import { stageFromEnvironment } from '../model/stage';
+import type {
+	CreatePaymentMethodState,
+	CreateSalesforceContactState,
+	User,
+	WrappedState,
+} from '../model/stateSchemas';
+import {
+	createPaymentMethodStateSchema,
+	wrapperSchemaForState,
+} from '../model/stateSchemas';
+import { ServiceProvider } from '../services/config';
+import { getPayPalConfig, PayPalService } from '../services/payPal';
+import { getStripeConfig, StripeService } from '../services/stripe';
+import { asRetryError } from '../util/errorHandler';
+import { getIfDefined } from '../util/nullAndUndefined';
+
+const stage = stageFromEnvironment();
+const stripeServiceProvider = new ServiceProvider(stage, async (stage) => {
+	const config = await getStripeConfig(stage);
+	return new StripeService(config);
+});
+const paypalServiceProvider = new ServiceProvider(stage, async (stage) => {
+	const config = await getPayPalConfig(stage);
+	return new PayPalService(config);
+});
+
+export const handler = async (
+	state: WrappedState<CreatePaymentMethodState>,
+) => {
+	try {
+		console.info(`Input is ${JSON.stringify(state)}`);
+		const createPaymentMethodState = wrapperSchemaForState(
+			createPaymentMethodStateSchema,
+		).parse(state).state;
+		return createSalesforceContactState(
+			state,
+			await createPaymentMethod(
+				createPaymentMethodState.paymentFields,
+				createPaymentMethodState.user,
+			),
+		);
+	} catch (error) {
+		throw asRetryError(error);
+	}
+};
+
+export function createPaymentMethod(
+	paymentFields: PaymentFields,
+	user: User,
+): Promise<PaymentMethod> {
+	switch (paymentFields.paymentType) {
+		case 'Stripe':
+			return createStripePaymentMethod(user.isTestUser, paymentFields);
+		case 'StripeHostedCheckout':
+			return createStripeHostedPaymentMethod(user.isTestUser, paymentFields);
+		case 'PayPal':
+			return createPayPalPaymentMethod(user.isTestUser, paymentFields);
+		case 'DirectDebit':
+			return createDirectDebitPaymentMethod(user, paymentFields);
+		case 'Existing':
+			return Promise.reject(
+				new Error(
+					'Existing payment methods should never make their way to this lambda',
+				),
+			);
+	}
+	return Promise.reject(new Error('Unknown payment method type'));
```

Comment:
Is it worth including the unknown type in the error? Might be helpful to know what it was if we ever reach this line.

Reply:
This is actually unreachable code, I'm surprised it wasn't being flagged by lint